relay-event-schema/src/protocol/ourlog.rs

```diff
@@ -1,243 +1,437 @@
-                map.insert("bool_value".to_string(), Annotated::new(Value::Bool(v)));
-            }
-            AttributeValue::Unknown(v) => {
-                map.insert("unknown".to_string(), Annotated::new(Value::String(v)));
-            }
+#[derive(Clone, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+pub struct OurLogAttribute {
+    #[metastructure(flatten)]
+    pub value: OurLogAttributeValue,
+
+    /// Additional arbitrary fields for forwards compatibility.
+    #[metastructure(additional_properties)]
+    pub other: Object<Value>,
+}
+
+impl OurLogAttribute {
+    pub fn new(attribute_type: OurLogAttributeType, value: Value) -> Self {
+        Self {
+            value: OurLogAttributeValue::new(Annotated::new(attribute_type), Annotated::new(value)),
+            other: Object::new(),
         }
-        Value::Object(map)
     }
+}
 
-    fn serialize_payload<S>(&self, s: S, _behavior: SkipSerialization) -> Result<S::Ok, S::Error>
-    where
-        Self: Sized,
-        S: serde::Serializer,
-    {
-        let mut map = s.serialize_map(None)?;
-        match self {
-            AttributeValue::StringValue(v) => {
-                map.serialize_entry("string_value", v)?;
-            }
-            AttributeValue::IntValue(v) => {
-                map.serialize_entry("int_value", v)?;
-            }
-            AttributeValue::DoubleValue(v) => {
-                map.serialize_entry("double_value", v)?;
-            }
-            AttributeValue::BoolValue(v) => {
-                map.serialize_entry("bool_value", v)?;
-            }
-            AttributeValue::Unknown(v) => {
-                map.serialize_entry("unknown", v)?;
-            }
+impl fmt::Debug for OurLogAttribute {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("OurLogAttribute")
+            .field("value", &self.value.value)
+            .field("type", &self.value.ty)
+            .finish()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+pub struct OurLogAttributeValue {
+    #[metastructure(field = "type", required = true, trim = false)]
+    pub ty: Annotated<String>,
+    #[metastructure(required = true, pii = "true")]
+    pub value: Annotated<Value>,
+}
+
+impl OurLogAttributeValue {
+    pub fn new(attribute_type: Annotated<OurLogAttributeType>, value: Annotated<Value>) -> Self {
+        Self {
+            ty: attribute_type.map_value(|at| at.as_str().to_string()),
+            value,
         }
-        map.end()
     }
 }
 
-impl AttributeValue {
-    pub fn string_value(&self) -> Option<&String> {
+#[derive(Debug)]
+pub enum OurLogAttributeType {
+    Bool,
+    Int,
+    Double,
+    String,
+    Unknown(String),
+}
+
+impl OurLogAttributeType {
+    fn as_str(&self) -> &str {
         match self {
-            AttributeValue::StringValue(s) => Some(s),
-            _ => None,
+            Self::Bool => "bool",
```

Comment:
```suggestion
            Self::Bool => "boolean",
```

cc @AbhiPrasad @Dav1dde